### PR TITLE
Add NeonEVM Mainnet and contracts

### PR DESCRIPTION
**Pull request by nick8319**
Issue: https://github.com/safe-global/safe-deployments/issues/267
Safe-deployments PR: https://github.com/safe-global/safe-deployments/pull/268

### CODE REVIEW

File: gnosis/eth/clients/etherscan_client.py [hunk 1]
```diff
@@ -44,7 +44,8 @@ class EtherscanClient:
         EthereumNetwork.CRONOS_TESTNET: "https://testnet.cronoscan.com",
         EthereumNetwork.CELO_MAINNET: "https://celoscan.io",
         EthereumNetwork.BASE_GOERLI_TESTNET: "https://goerli.basescan.org",
-        EthereumNetwork.NEON_EVM_DEVNET: "https://neonscan.org",
+        EthereumNetwork.NEON_EVM_DEVNET: "https://devnet.neonscan.org",
```
Comment (moisses89) [inline gnosis/eth/clients/etherscan_client.py:47]: devnet url explorer does not work. It is a temporary issue?

Comment (moisses89) [reply gnosis/eth/clients/etherscan_client.py:47]: It works again

Comment (nick8319) [reply gnosis/eth/clients/etherscan_client.py:47]: Looks like it was a temporary issue

File: gnosis/safe/addresses.py [hunk 2]
```diff
@@ -893,4 +901,10 @@
     EthereumNetwork.OASIS_SAPPHIRE: [
         ("0xC22834581EbC8527d974F8a1c97E1bEA4EF910BC", 325632),  # v1.3.0
     ],
+    EthereumNetwork.NEON_EVM_DEVNET: [
+        ("0xC22834581EbC8527d974F8a1c97E1bEA4EF910BC", 205146874),  # v1.3.0
+    ],
+    EthereumNetwork.NEON_EVM_MAINNET: [
+        ("0xC22834581EbC8527d974F8a1c97E1bEA4EF910BC", 203993869),  # v1.3.0
```
Comment (moisses89) [inline gnosis/safe/addresses.py:908]: This block is not having transactions, can you double check?
https://neonscan.org/block/203993869#transactions

**Review by moisses89 — CHANGES_REQUESTED** [review]

File: gnosis/safe/addresses.py (hunk 2 above)
Comment (nick8319) [reply gnosis/safe/addresses.py:908]: Well, it has 1 tx - https://neonscan.org/tx/0x13823780528d1281bd802a8744bae19f14b62f941f7fdf936b1b67a4c85b92a1
Probably again was an explorer temporary glitch

**Comment by moisses89** [comment]
@nick8319 can you solve the conflicts?

**Comment by ElvisKrop** [comment]
Hello @moisses89!
Conflicts are solved :heavy_check_mark: 
Please, merge this PR asap.
Thank you! Have a nice day and week ahead!

**Pull request merged**